src/config/cargo_crates_toml.rs

```diff
@@ -81,57 +82,92 @@ impl CargoCratesToml {
     /// to vector of binary names and return a couple of package name to
     /// [`Package`] information.
     ///
-    /// The current crate will be kept in the packages iff `keep_self` is
+    /// The current crate will be kept in the packages if `keep_self` is
     /// `true`, otherwise it will be filtered out.
-    fn into_config<F>(self, pkg_map: F, keep_self: bool) -> UserConfig
+    ///
+    /// All locally-installed packages are kept if `with_local` is
+    /// `true`, otherwise they will be filtered out.
+    fn into_config<F>(self, pkg_map: F, keep_self: bool, with_local: bool) -> UserConfig
```

Comment:
If you could also rename all these `with_local`s to `keep_local`, it would be really nice. `grep -r with_local` to be sure you didn't miss anything.